bgpd/bgp_nht.c

```diff
@@ -896,9 +896,12 @@ static void sendmsg_zebra_rnh(struct bgp_nexthop_cache *bnc, int command)
 	ret = zclient_send_rnh(zclient, command, &bnc->prefix, exact_match,
 			       resolve_via_default, bnc->bgp->vrf_id);
 	/* TBD: handle the failure */
-	if (ret == ZCLIENT_SEND_FAILURE)
+	if (ret == ZCLIENT_SEND_FAILURE) {
 		flog_warn(EC_BGP_ZEBRA_SEND,
 			  "sendmsg_nexthop: zclient_send_message() failed");
+		UNSET_FLAG(bnc->flags, BGP_NEXTHOP_REGISTERED);
```

Comment:
What if the nexthop is currently registered and the command is `ZEBRA_NEXTHOP_UNREGISTER`?
You're doing the same thing that is done when we have a successful command. I think when we have a failure, we don't need to touch any flag and should just return.

Reply:
BTW, I think the TBD comment above should be removed if you do this change.

Reply:
Thanks for your comment！